packages/database/src/migrations/20210816012839-AddProjectPacmossi-modifies-data.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+import { codeToId, insertObject, generateId, findSingleRecord, deleteObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const countryCodes = ['FJ', 'KI', 'NR', 'NU', 'PG', 'SB', 'TV', 'TO', 'VU'];
+
+const projectCode = 'pacmossi';
+const projectName = 'PacMOSSI';
+const projectDescription = 'Pacific Mosquito Surveillance Strengthening for Impact';
+
+const userGroups = ['PacMOSSI', 'PacMOSSI Senior'];
+
+const dashboardName = 'Vector Surveillance';
+const projectDashboardItemCode = 'project_details';
+const projectDashboardCode = `${projectCode}_${projectName}`;
+
+const getCountryDashboardCode = (countryCode, userGroup) =>
+  `${countryCode}_PacMOSSI_Country_${userGroup.split(' ').join('_')}`;
+
+const hierarchyNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const addCountryToProject = async (db, countryCode) => {
+  await insertObject(db, 'entity_relation', {
+    id: generateId(),
+    parent_id: await codeToId(db, 'entity', projectCode),
+    child_id: await codeToId(db, 'entity', countryCode),
+    entity_hierarchy_id: await hierarchyNameToId(db, projectCode),
+  });
+  await Promise.all(
+    userGroups.map(userGroup =>
+      insertObject(db, 'dashboard', {
+        id: generateId(),
+        code: getCountryDashboardCode(countryCode, userGroup),
+        name: dashboardName,
+        root_entity_code: countryCode,
+        sort_order: 1,
+      }),
+    ),
+  );
+};
+
+const addItemToDashboard = async (
+  db,
+  { code, dashboardCode, permissionGroup, entityTypes, projectCodes },
+) => {
+  const dashboardItemId = (await findSingleRecord(db, 'dashboard_item', { code })).id;
+  const dashboardId = (await findSingleRecord(db, 'dashboard', { code: dashboardCode })).id;
+  await insertObject(db, 'dashboard_relation', {
+    id: generateId(),
+    dashboard_id: dashboardId,
+    child_id: dashboardItemId,
+    entity_types: `{${entityTypes.join(', ')}}`,
+    project_codes: `{${projectCodes.join(', ')}}`,
+    permission_groups: `{${permissionGroup}}`,
+    sort_order: 1,
+  });
+};
+
+exports.up = async function (db) {
+  await insertObject(db, 'entity', {
+    id: generateId(),
+    code: projectCode,
+    parent_id: await codeToId(db, 'entity', 'World'),
+    name: projectName,
+    type: 'project',
+  });
+
```

Comment:
either 
```suggestion
const entityHierarchyId = generateId();
 
```
or you could use the following fetch after insert:
```suggestion
const entityHierarchyId =await hierarchyNameToId(db, projectCode);
```
I'd probably go the former.